src/highlight-support.js

```diff
@@ -45,8 +45,9 @@ const highlightSupport = {
       this.win
     )
 
-    const actualStartIndex = startIndex
-    const actualEndIndex = endIndex
+    // Do not let highlight exceed text range
+    const actualStartIndex = Math.max(startIndex, 0)
+    const actualEndIndex = Math.min(endIndex, blockText.length)
```

Comment:
These values could also be clamped, so that the start cannot be greater than one character before the end of the string, and the end cannot be less than 1.